File: week1-python/Image_Annotation/Drawing_Ellipse.py
# Import cv2 module
import cv2
from dataPath import DATA_PATH

# Path to the image we are going to read
# This can be an absolute or relative path
# Here we are using a relative path
imageName = DATA_PATH+"images/boy.jpg"

# Load the image
image = cv2.imread(imageName, cv2.IMREAD_COLOR)

# Draw an ellipse
# Note: Ellipse Centers and Axis lengths must be integers
imageEllipse = image.copy()
cv2.ellipse(imageEllipse, (250, 125), (100, 50), 0, 0, 360, (255, 0, 0), thickness=3, lineType=cv2.LINE_AA);
cv2.ellipse(imageEllipse, (250, 125), (100, 50), 90, 0, 360, (0, 0, 255), thickness=3, lineType=cv2.LINE_AA);

# Display the image
#cv2.imshow("Image Ellipse",imageEllipse)
#cv2.waitKey(0)
cv2.imwrite("../results/imageEllipse.png",imageEllipse)

# Draw an ellipse
# Note: Ellipse Centers and Axis lengths must be integers
imageEllipse = image.copy()
# Incomplete/Open ellipse
cv2.ellipse(imageEllipse, (250, 125), (100, 50), 0, 180, 360, (255, 0, 0), thickness=3, lineType=cv2.LINE_AA);
# Filled ellipse
cv2.ellipse(imageEllipse, (250, 125), (100, 50), 0, 0, 180, (0, 0, 255), thickness=-2, lineType=cv2.LINE_AA);

# Display the image
#cv2.imshow("Image Filled Ellipse",imageEllipse)
#cv2.waitKey(0)
cv2.imwrite("../results/imageFilledEllipse.png",imageEllipse)

File: week1-cpp/Image_Annotation/Drawing_Ellipse.cpp
// Include libraries
#include <iostream>
#include "dataPath.hpp"
#include <opencv2/core.hpp>
#include <opencv2/imgproc.hpp>
#include <opencv2/highgui.hpp>
using namespace std;
using namespace cv;

int main(void){

        // Path to the image we are going to read
        // This can be an absolute or relative path
        // Here we are using a relative path
        string imageName = DATA_PATH+"images/boy.jpg";

        // Load the image
        Mat image;
        image = imread(imageName, IMREAD_COLOR);

	// Draw an ellipse
	// Note: Ellipse Centers and Axis lengths must be integers
	Mat imageEllipse = image.clone();

	ellipse(imageEllipse, Point(250, 125), Point(100, 50), 0, 0, 360,
        Scalar(255, 0, 0), 3, LINE_AA);

	ellipse(imageEllipse, Point(250, 125), Point(100, 50), 90, 0, 360,
        Scalar(0, 0, 255), 3, LINE_AA);

	imwrite("../results/ellipse.jpg",imageEllipse);
	//imshow("Draw ellipse on image", imageEllipse);
	//waitKey(0);
	
	// Draw an ellipse
	// Note: Ellipse Centers and Axis lengths must be integers
	imageEllipse = image.clone();

	// Incomplete/Open ellipse
	ellipse(imageEllipse, Point(250, 125), Point(100, 50), 0, 180, 360,
        Scalar(255, 0, 0), 3, LINE_AA);

	// Filled ellipse
	ellipse(imageEllipse, Point(250, 125), Point(100, 50), 0, 0, 180,
        Scalar(0, 0, 255), -2, LINE_AA);

	imwrite("../results/ellipseFilled.jpg",imageEllipse);

	//imshow("Draw filled ellipse",imageEllipse);
	//waitKey(0);
	return 0;
}
